pkg/kn/commands/wait_flags.go

```diff
@@ -0,0 +1,43 @@
+package commands
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+)
+
+// Flags for tuning wait behaviour
+type WaitFlags struct {
+	Timeout int
+
+	// Don't access directly, use IsWait()
+	noWaitFlag bool
+	waitFlag   bool
+}
+
+// Add flags which influence the sync/async behaviour when creating or updating
+// resources. Set `waitDefault` argument if the default behaviour is synchronous.
+// Use `what` for describing what is waited for.
+func (p *WaitFlags) AddWaitFlags(command *cobra.Command, waitDefault bool, waitTimeoutDefault int, what string) {
```

Comment:
i dont think we need waitDefault

Reply:
I probably went a bit too far here. The idea was that the default behaviour (sync/async) could be configured in configuration file, but on second thought that is not a good idea. A command line should be completely self-container without referring to any hidden state (so that e.g. c&p works always).

I remove that.